src/SystemReport.php

```diff
@@ -0,0 +1,248 @@
+		return $output;
+	}
+
+	/**
+	 * Exclude specific settings from the system report.
+	 *
+	 * - if you pass an array, the 'key' is the form field key, and the 'value' is the value of that form field who you want to match against.
+	 * - if you pass a string, it will match the setting option name.
+	 * - you may mix both strings and arrays.
+	 *
+	 * @example `'key' => 'title', 'value' => 'enable/disable'` will exclude all form fields whose title is 'enable/disable'.
+	 *
+	 * @param array $settings The settings to exclude.
+	 */
+	public function exclude( $settings ) {
+		$this->excluded = $settings;
+	}
+
+	/**
+	 * Include ONLY specific settings in the system report.
+	 *
+	 * - if you pass an array, the 'key' is the form field key, and the 'value' is the value of that form field who you want to match against.
+	 * - if you pass a string, it will match the setting option name.
+	 * - you may mix both strings and arrays.
+	 *
+	 * @example `'key' => 'type', 'value' => 'checkbox'` mean include all checkbox settings.
+	 *
+	 * @param array $settings The settings to include.
+	 */
+	public function include( $settings ) {
+		$this->included = $settings;
+	}
+
+	/**
+	 * Displays the log entries on the System Report page.
+	 *
+	 * @return void
+	 */
+	public function add_status_page_box() {
+		$settings = $this->get_current_settings();
+
+		$id   = $this->id;
+		$name = $this->name;
+		include_once __DIR__ . '/includes/admin/views/status-report.php';
+	}
+
+	/**
+	 * Add a log entry to the system report.
+	 *
+	 * @template T
+	 * @param T     $response The API request that you want to report about.
+	 * @param mixed $extra    Any extra information you want to include in the report.
+	 *
+	 * @return T
+	 */
+	public function request( $response, $extra = null ) {
+		if ( ! is_wp_error( $response ) ) {
+			return $response;
+		}
+
+		$logs   = json_decode( get_option( 'krokedil_support_' . $this->id, '[]' ), true );
+		$logs[] = array(
+			'timestamp' => current_time( 'mysql' ),
+			'response'  => array(
+				'code'    => $response->get_error_code(),
+				'message' => $response->get_error_message(),
+				'extra'   => $extra,
+			),
+		);
+
+		update_option( 'krokedil_support_' . $this->id, wp_json_encode( $logs ) );
+		return $response;
+	}
+
+	/**
+	 * Remove old report entries.
+	 *
+	 * @hook woocommerce_cleanup_logs
+	 */
+	public static function remove_old_entries() {
+		$registry = get_option( self::REGISTRY_OPTION, array() );
+		foreach ( $registry as $id ) {
+			$reports = json_decode( get_option( 'krokedil_support_' . $id, '[]' ), true );
+			if ( empty( $reports ) ) {
+				continue;
+			}
+
+			$retention_period = wc_get_container()->get( Settings::class )->get_retention_period();
+			foreach ( $reports as $report ) {
+				if ( strtotime( $report['timestamp'] ) < strtotime( "-{$retention_period} days" ) ) {
+					unset( $reports[ $report ] );
+				}
+			}
+			update_option( 'krokedil_support_' . $id, wp_json_encode( $reports ) );
+		}
+	}
```

Comment:
Change this to not be a static method, and to skip the loop for the registry. Since each plugin will create a instance of this class the cleanup logs action will trigger each plugins class instance anyway with their own id set in the constructor.